Sources/Signal.swift

```diff
@@ -127,28 +119,20 @@ public final class Signal<Value, Error: Swift.Error> {
 				}
 
 				self.sendLock.unlock()
+			}
 
-				// Check if the status has been bumped to `terminating` due to a
-				// terminal event being sent concurrently or recursively.
-				//
-				// The check is deliberately made outside of the `sendLock` so that it
-				// covers also any potential concurrent terminal event in one shot.
-				//
-				// Related PR:
-				// https://github.com/ReactiveCocoa/ReactiveSwift/pull/112
-				self.stateLock.lock()
-				if result == .none, case .terminating = self.state {
-					self.stateLock.unlock()
-					result = self.tryToCommitTermination()
```

Comment:
`tryToCommitTermination` already locks the state to check if it is terminating or not. So the locking here is pointless.